cylc/flow/task_pool.py

```diff
@@ -768,6 +827,12 @@ def remove(self, itask, reason=""):
         if reason:
             msg += f" ({reason})"
 
+        if itask.is_xtrigger_sequential:
+            with suppress(ValueError):
+                self.xtrigger_mgr.sequential_spawn_next.remove(
+                    itask.identity
+                )
```

Comment:
Why would a `ValueError` be raised here?
```suggestion
            self.xtrigger_mgr.sequential_spawn_next.remove(itask.identity)
```
or, if you expect a `KeyError` to be raised, use `set.discard()`:
```suggestion
            self.xtrigger_mgr.sequential_spawn_next.discard(itask.identity)
```